### ISSUE TITLE
Does not work with detached custom elements

### ISSUE THREAD

**Issue by dy** [atomicojs/atomico#15]
If we'd like to create and run custom element in detached tree, it is impossible to run atomico handler with it.
```js
import { customElement, useState } from 'atomico'
function MyTag() {
  let [ count, setCount ] = useState(0)
  setTimeout(() => {
    console.log(count)
    setCount(++count)
  }, 1000)
  return {nodeType: 'host'}
}
customElement("my-tag", MyTag);
let el = document.createElement('my-tag')
```
This just silently ignores the component.

If we try to render the old scool react way
```js
import { ... , render } from 'atomico'
...
render(MyTag, document.createDocumentFragment())
```
It gives error - hooks are not supported.

[haunted](https://ghub.io/haunted) doesn't have that drawback. Any plans to catch up?

**Comment by dy** [comment]
(I was trying to include atomico as target in [enhook](https://github.com/dy/enhook) - I wonder if that's possible at all to create atomico component without DOM).

**Comment by UpperCod** [comment]
Thank you for your example, I have detected an interesting problem:

Atomico and Haunted work similarly wait for the component to be mounted, to render hooks and views, but currently differently when unmounted.

Haunted follows the setTimeout buble, Atomico pause it, [see example](https://codepen.io/uppercod/pen/GRRXGVL?editors=1010)

Atomico in version 0.13.1 correctly reintegrates the cycle(**This is thanks to your observation** @dy ), but it would be ideal if haunted could pause the render cycle to avoid memory loss and additional DOM work...

**Comment by dy** [comment]
haunted exposes State class so that it can be run independent of DOM. Is there something similar in atomico?

**Comment by UpperCod** [comment]
Yes, it exists undocumented and interesting. [atomico/src/core/hooks.js](https://github.com/atomicojs/atomico/blob/master/src/core/hooks.js)

```js
import { createHookCollection, useState } from "atomico";

/**
 *  createHookCollection, Create a collection of hooks, it
 * must be stored for reuse if you want to maintain the state between renders
 */
let hooks = createHookCollection(() => {
  /**
   * any render
   **/
});

/**
 * function that uses Atomic hooks
 */
function Example({ any }) {
  let [count, setCount] = useState(0);
}

/**
 * open the hooks, marking positions
 */
hooks.load(Example, { any: "param" });

/**
 * dispatch the updated life cycle
 */
hooks.updated();

/**
 * dispatch the life cycle unmount
 */
hoooks.unmount();

```

> To implement a customRender outside of Atomico you will need to save the variable `hooks` of the example and then retrieve it, Atomico saves it in each [instance of a web-component](https://github.com/atomicojs/atomico/blob/master/src/core/element/element.js#L63)

In the file [atomico/src/core/hooks.js](https://github.com/atomicojs/atomico/blob/master/src/core/hooks.js), internal hooks are used as:

`useRender`, get the render associated with the hook
 `useHook`, It is a pattern to reducer, to know the life cycle of the hook and modify its state, [eg useEffect](https://github.com/atomicojs/atomico/blob/master/src/core/hooks.js#L92-L126)

**All this within the 3KB of Atomico**

**Comment by dy** [comment]
Hm. I'm trying to 
```js
import { createHookCollection, useState, useEffect } from 'atomico'

function myFn() {
  let [count, setCount] = useState(0)
  useEffect(() => setCount(1), [])
  hooks.updated()
}

let hooks = createHookCollection(() => { })
hooks.load(myFn)
myFn()
```

And getting error:
```js
undefined is not iterable
```
The `useState(0)` returns `undefined` during the second call.
What am I getting wrong here?

**Comment by matthewp** [comment]
Hey just popping in here, I'm the maintainer of Haunted. Just wanted to say that the unmounted issue with unmounted components is an interesting one and worth discussing further. If anyone wants to open an issue in matthewp/haunted we can discuss there, I don't want to sidetrack this issue. Thanks!

**Comment by UpperCod** [comment]
I am attentive to its implementation [👀](https://github.com/dy/enhook/blob/master/src/provider/atomico.js), you should create a layered function to encapsulate the hook collection and retrieve it, eg

[https://codepen.io/uppercod/pen/ZEEqWEg?editors=0012](https://codepen.io/uppercod/pen/ZEEqWEg?editors=0012)

```js
import { createHookCollection,  useState } from "atomico";
/**
 * store createHookCollection
 */
let components = new Map();

/**
 * create a scope for hooks
 * @param {Function} component
 * @return {Function} render
 */
function createComponent(component) {
  let hooks = createHookCollection(render);
  /**
   * custom render that allows you to use Atomico hooks
   * @param {any} props
   */
  function render(props) {
    hooks.load(component, props);
    hooks.updated();
  }
  /**
   * save the createHookCollection
   */
  components.set(component, hooks);

  return render;
}

/**
 *  The return will be the render function to insert the component.
 */
let renderComponent = createComponent(() => {
  let [state, setState] = useState(0);
  setTimeout(() => {
    setState(state + 1);
  }, 1000);

  console.log(state);
});

renderComponent({ any: "prop" });
```

**Comment by dy** [comment]
Hmm. `useEffect` creates stackoverflow in that example @UpperCod: https://codepen.io/dyv/pen/BaaqKLO?editors=1111

**Comment by UpperCod** [comment]
Fixed in version atomico@0.13.3, please check the link  https://codepen.io/uppercod/pen/eYYPWOq?editors=0012

**Comment by dy** [comment]
Nice. Added enhook implementation https://github.com/unihooks/enhook/blob/master/src/provider/atomico.js
